design/export-functionality.md

```diff
@@ -0,0 +1,85 @@
+# Export
+
+## Requirements
+The functionality currently provided by the [lakeFS Export utility](https://docs.lakefs.io/reference/export.html) but without a dependency on Apache Spark.
+This includes:
+1. Export all data from a given lakeFS reference (commit or branch) to a designated object store location.
+1. Export only the diff between the HEAD of a branch and a commit reference (helpful for continuous exports).
+1. Success/Failure indications on the exported objects. 
+
+
+## Non-Requirements
+1. Support distribution across machines.
+1. The functionality doesn't need to be supported from lakeFS.
+
+
+## Possible Solutions external to lakeFS
+
+### Spark client stand-alone
+
+Create a standalone Docker container that runs the spark client.
+
+In order to use Docker containers, users will need to download it if necessary, learn it, and have it in their toolset.
+
+Usage example:
+
+`docker run lakefs-export --conf spark.hadoop.lakefs.api.url=https://<LAKEFS_ENDPOINT>/api/v1 \
+--conf spark.hadoop.lakefs.api.access_key=<LAKEFS_ACCESS_KEY_ID> \
+--conf spark.hadoop.lakefs.api.secret_key=<LAKEFS_SECRET_ACCESS_KEY> \
+--packages io.lakefs:lakefs-spark-client-301_2.12:0.1.0 \
+--class io.treeverse.clients.Main export-app example-repo s3://example-bucket/prefix \
+--branch=example-branch`
```

Comment:
```suggestion
```docker run lakefs-export --conf spark.hadoop.lakefs.api.url=https://<LAKEFS_ENDPOINT>/api/v1 \
--conf spark.hadoop.lakefs.api.access_key=<LAKEFS_ACCESS_KEY_ID> \
--conf spark.hadoop.lakefs.api.secret_key=<LAKEFS_SECRET_ACCESS_KEY> \
--packages io.lakefs:lakefs-spark-client-301_2.12:0.1.0 \
--class io.treeverse.clients.Main export-app example-repo s3://example-bucket/prefix \
--branch=example-branch```
```

Reply:
adding to the above - use ```shell
in order to have syntax highlight for the docker run command